bullet-featherstone/src/JointFeatures.cc

```diff
@@ -341,6 +341,31 @@ Identity JointFeatures::AttachFixedJoint(
   if (world != nullptr && world->world)
   {
     world->world->addMultiBodyConstraint(jointInfo->fixedConstraint.get());
+
```

Comment:
I tried adding the following here
```
    jointInfo->fixedConstraint->setMaxAppliedImpulse(btScalar(1e9));
```
and disabling `fixedConstraintWeldChildToParent` in Base.hh. By default the max impulse is 100. Setting a high value like this made it so that the constraint is satisfied with minimal oscillations. As we discussed, the transform tool does not apply forces on the bodies, so the resulting behavior cannot be relied upon to determine if the physics is correct. Instead, I set up a simulation where two boxes are attached by a detachable joint and the parent box is attached to a fixed base with a prismatic joint.
<details><summary>Test SDFormat file</summary>

```xml
<?xml version="1.0"?>
<sdf version="1.11">
  <world name="default">
  <light type="directional" name="sun">
      <cast_shadows>true</cast_shadows>
      <pose>0 0 10 0 0 0</pose>
      <diffuse>1 1 1 1</diffuse>
      <specular>0.5 0.5 0.5 1</specular>
      <attenuation>
        <range>1000</range>
        <constant>0.9</constant>
        <linear>0.01</linear>
        <quadratic>0.001</quadratic>
      </attenuation>
      <direction>-0.5 0.1 -0.9</direction>
    </light>

    <model name="ground_plane">
      <static>true</static>
      <link name="link">
        <collision name="collision">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
            </plane>
          </geometry>
        </collision>
        <visual name="visual">
          <geometry>
            <plane>
              <normal>0 0 1</normal>
              <size>100 100</size>
            </plane>
          </geometry>
          <material>
            <ambient>0.8 0.8 0.8 1</ambient>
            <diffuse>0.8 0.8 0.8 1</diffuse>
            <specular>0.8 0.8 0.8 1</specular>
          </material>
        </visual>
      </link>
    </model>
    <model name="M1">
      <joint name="j_base" type="fixed">
        <parent>world</parent>
        <child>base</child>
      </joint>
      <link name="base">
        <pose>0 0 0.05  0 0 0</pose>
        <inertial auto="true"/>
        <collision name="box_collision">
          <density>1000</density>
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
        </collision>
        <visual name="box_visual">
          <geometry>
            <box>
              <size>0.1 0.1 0.1</size>
            </box>
          </geometry>
          <material>
            <ambient>1 0 0 0.1</ambient>
            <diffuse>1 0 1 0.1</diffuse>
            <specular>1 0 0 1</specular>
          </material>
        </visual>
      </link>
      <joint name="j1" type="prismatic">
        <parent>base</parent>
        <child>body</child>
        <axis>
          <xyz>0 1 0</xyz>
        </axis>
      </joint>
      <plugin
        filename="gz-sim-joint-position-controller-system"
        name="gz::sim::systems::JointPositionController">
        <joint_name>j1</joint_name>
        <p_gain>20000</p_gain>
        <d_gain>10000</d_gain>
        <cmd_max>10000</cmd_max>
        <cmd_min>-10000</cmd_min>
        <use_velocity_commands>true</use_velocity_commands>
        <topic>cmd_pos</topic>
      </plugin>
      <link name="body">
        <pose>1 0 0.25 0 0.0 0.0</pose>
        <inertial auto="true"/>
        <collision name="box_collision">
          <density>100</density>
          <geometry>
            <box>
              <size>1 1 0.5</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>0.1</mu>
                <mu2>0.1</mu2>
              </ode>
              <bullet>
                <friction>0.1</friction>
                <friction2>0.1</friction2>
              </bullet>
            </friction>
          </surface>
        </collision>

        <visual name="box_visual">
          <geometry>
            <box>
              <size>1 1 0.5</size>
            </box>
          </geometry>
          <material>
            <ambient>1 0 0 1</ambient>
            <diffuse>1 0 0 1</diffuse>
            <specular>1 0 0 1</specular>
          </material>
        </visual>
      </link>
      <plugin filename="gz-sim-detachable-joint-system"
              name="gz::sim::systems::DetachableJoint">
       <parent_link>body</parent_link>
       <child_model>M2</child_model>
       <child_link>body2</child_link>
      </plugin>
    </model>

    <model name="M2">
      <pose>3.3 0 0.25 0 0.0 0</pose>
      <link name="body2">
        <inertial auto="true"/>
        <collision name="box_collision">
          <density>100</density>
          <geometry>
            <box>
              <size>1 1 0.5</size>
            </box>
          </geometry>
          <surface>
            <friction>
              <ode>
                <mu>1</mu>
                <mu2>1</mu2>
              </ode>
              <bullet>
                <friction>1</friction>
                <friction2>1</friction2>
              </bullet>
            </friction>
          </surface>
        </collision>

        <visual name="box_visual">
          <geometry>
            <box>
              <size>1 1 0.5</size>
            </box>
          </geometry>
          <material>
            <ambient>0 1 0 1</ambient>
            <diffuse>0 1 0 1</diffuse>
            <specular>1 0 0 1</specular>
          </material>
        </visual>
      </link>
    </model>
  </world>
</sdf>

```

</details> 

 Here are the results:

* **DART**
[dart.webm](https://github.com/gazebosim/gz-physics/assets/206116/c0025dce-20e4-4648-a998-2f96932fd47e)

* **Bullet-featherstone gz-physics7 branch**
[bullet-default.webm](https://github.com/gazebosim/gz-physics/assets/206116/36ee957d-493b-4b11-b40f-7ce359049368)

* **Bullet-featherstone current PR**
[current-PR.webm](https://github.com/gazebosim/gz-physics/assets/206116/f8d4995f-cd2d-4a1b-8b75-5a323387a99d)

* **Bullet-featherstone proposed change**: setting a high max impulse
[max_impulse.webm](https://github.com/gazebosim/gz-physics/assets/206116/497c6458-0a25-4cdb-96d4-724b94825eea)

I moved your changes in SimulationFeatures.cc directly into `FreeGroupFeatures::SetFreeGroupWorldPose` and that solved the transform tool issue.

Reply:
ok thanks for testing. I revised this PR based on the suggestions. Changes are:
*  added `jointInfo->fixedConstraint->setMaxAppliedImpulse(btScalar(1e9));` when creating attaching joint
* moved code for enforcing child body pose  to `SetFreeGroupWorldPose` and updated test
* removed `SetFixedJointWeldChildToParent feature`